ansible_runner/runner_config.py

```diff
@@ -724,12 +724,30 @@ def _parse_cli_execenv_cmd_playbook_args():
 
             if 'podman' in self.process_isolation_executable:
                 # container namespace stuff
+                new_args.extend(["--group-add=root"])
                 new_args.extend(["--userns=keep-id"])
                 new_args.extend(["--ipc=host"])
 
 
-        # Mount the private_data_dir
-        new_args.extend(["-v", "{}:/runner:Z".format(self.private_data_dir)])
+        # These directories need to exist before they are mounted in the container,
+        # or they will be owned by root.
+        private_subdirs = [
+            d for d in os.listdir(self.private_data_dir) if os.path.isdir(
+                os.path.join(self.private_data_dir, d)
+            )
+        ]
+
+        if 'artifacts' not in private_subdirs:
+            private_subdirs += ['artifacts']
+
+        for d in private_subdirs:
+            if not os.path.exists(os.path.join(self.private_data_dir, d)):
+                if d == 'artifacts':
+                    os.mkdir(os.path.join(self.private_data_dir, d), 0o700)
```

Comment:
This should _only_ apply to artifacts, because `private_subdirs` only includes things which `os.path.isdir`

so you could write

```python
          if not os.path.exists(os.path.join(self.private_data_dir, 'artifacts')):
                os.mkdir(os.path.join(self.private_data_dir, 'artifacts'), 0o700)
```

This is what some of my closed PRs looked like